src/lib.rs

```diff
@@ -1,40 +1,33 @@
-/// Firezone explicitly ties the WireGuard tunnel interface lifetime to
-/// the Control place connection lifetime under a single struct called Session.
+use platform::tunnel::Tunnel;
+
+mod platform;
+
+#[allow(dead_code)]
 pub struct Session {
-    fd: u32,
-    handle: u32,
-    // TODO: WebSocket connection handle etc
+    tunnel: Tunnel,
 }
 
 impl Session {
-    pub fn connect(_portal_url: String, _token: String) -> Session {
-        // TODO: Do this in a more platform agnostic way
-        let fd = 9999;
-        let handle = fd + 1;
-
-        // 1. Find a free file descriptor (UNIX)
-        // 2. Allocate a device handle (UNIX)
-        // 3. Save this mapping for looking up later
-        // 4. Connect the WebSocket
-        println!("Found a free file descriptor: {fd}");
-        println!("Allocated device handle: {handle}");
-
-        // TODO: Mock websocket connection, periodically returning updated resources to the client
-        println!("Connected to portal");
-
-        Session { fd, handle }
+    pub fn connect(_portal_url: String, _token: String) -> Result<Session, std::io::Error> {
```

Comment:
just a a side note we will want to write our own error types and use that instead of `std::io::Error` here but we can leave that for later.

Reply:
Noted, yeah the types will need a lot of work I'm sure